### DISCUSSION TITLE
Alfa AWUS 1900 won't work

### DISCUSSION THREAD

**Discussion by Tockman** [chris1111/Wireless-USB-Big-Sur-Adapter#130] [General]
I've just installed Wireless USB Big Sur Adapter V9 on MacOS 12 Monterey b3. Installation went smooth and obviously successful (SIP is disabled). I gave permission to install third party kext, plugged adapter's USB connector in and rebooted. After that I see utility icon in menu, and system indicates USB connection (when I plug adapter's USB cable in and out) but doesn't recognize adapter itself. In System > Network > 802.11ac NIC tab I can see the status: "Cable for 802.11ac NIC isn't plugged or device doesn't respond". Looks like no proper driver (RTL8814) was installed. Is there any solution? Thank you again, Chris.

**Comment by Tockman** [comment]
NIC is recognized as USB2.0 device, not USB3.0 one. Both RtWlanU1827.kext and RtWlanU.kext are there in /Library/Extensions.

**Reply by Tockman** [reply]
Utility in Menu shows networks available but never able to connect to them.

**Reply by Tockman** [reply]
Special program for Alfa AWUS036ACH doesn't work for me - it not only shows NIC not plugged but even never finds available networks.

**Comment by chris1111** [comment]
Uninstall  reboot then use latest release

**Reply by Tockman** [reply]
Just a dumb control question: if I'm on an unsupported Mac using Monterey with OCLP, should I use "OC" or "non-OC" version of utility? Is that a Hackintosh, technically?

**Comment by chris1111** [comment]
for non-OC! you suppose to have this message

![image](https://user-images.githubusercontent.com/6248794/127742405-d759c9c6-f2a4-445e-af4f-a81c5d02ed33.png)
![image](https://user-images.githubusercontent.com/6248794/127742360-a7081789-8600-4b66-b6ed-cd3386324043.png)

**Reply by Tockman** [reply]
Yes, I had all of that. And result was as described: utility seeing available networks but OS not seeing NIC as connected.

**Reply by Tockman** [reply]
I guess, something might go wrong at the stage of pre-boot volume update. Will try one more time...

**Reply by Tockman** [reply]
No luck. And indicator on adapter even doesn't turn on.
![Снимок экрана 2021-07-31 в 18 11 00](https://user-images.githubusercontent.com/74548419/127744368-1c05553e-ec74-4fcb-ba0c-4676a6bb40cc.png)

**Reply by chris1111** [reply]
to turn on you need activate with pasword here

![127744368-1c05553e-ec74-4fcb-ba0c-4676a6bb40cc](https://user-images.githubusercontent.com/6248794/127744507-6a19072a-d52a-40fe-9fbe-7fe355daacd3.png)

<img width="402" alt="Bar wifi" src="https://user-images.githubusercontent.com/6248794/127744546-4e83eade-2db0-4f25-8686-7ebd288e82a6.png">

**Reply by Tockman** [reply]
Sure, I know it and I do it. It's my network, and it's password protected and WPA-enabled ;) After the password is entered utility tries to connect for some time, gets no success, and then icon in Menu changes from 'wifi' symbol back to 'device'.

**Reply by Tockman** [reply]
But the main question is - why doesn't my NIC get recognized by MacOS ("cable is disconnected")?

**Comment by chris1111** [comment]
So this is a Hardware issue probably relate to **the Old mac patching** try the adapter on other mac or Hackintosh and I almost certain it will work

**Reply by Tockman** [reply]
I installed it on my Windows, and it turned on and functioned normally. It's a pity - I bought it especially for Mac work...

**Reply by chris1111** [reply]
> I installed it on my Windows, and it turned on and functioned normally. It's a pity - I bought it especially for Mac work...


try the adapter on other mac or Hackintosh and I almost certain it will work

**Reply by Tockman** [reply]
Well, thank you but I need it exactly on this Mac.

**Reply by Tockman** [reply]
The last expectation is it could be USB-related Monterey problem.

**Reply by Tockman** [reply]
Chris, I need some help. Excuse me please for such kind of apply. Would you make a special program for my 1900 if I gave you original RtWlanU.kext and RtWlanU1827.kext from last official MacOS 10.15 driver? On your conditions, of course.

**Reply by Tockman** [reply]
Looks like it's not a hardware problem. Windows where I successfully installed my AWUS, lives in virtual machine on the same Mac. So, if USB ports are working OK in VR then they are normally working in general, right?

**Reply by chris1111** [reply]
1900 ?

**Reply by Tockman** [reply]
Yes. Alfa AWUS 1900. Just like in title.

**Reply by Tockman** [reply]
Still bad luck :(( Utility loads and shows all available networks. In System > Network - still "Cable unplugged, or device doesn't respond". Adapter doesn't turn on...
Thanks a lot for your effort and readiness to help, Chris!

**Comment by chris1111** [comment]
Its problematic with Alpha adapter  here

**Comment by chris1111** [comment]
I told you I am prety sur problem is come from the old mac Patching system

**Reply by Tockman** [reply]
I see. But may be there's some way to overcome it...

**Reply by chris1111** [reply]
I never have a  old mac patching system I can not help you

**Comment by chris1111** [comment]
Try this release
[Wireless.USB.Big.Sur.Adapter-V-4.zip](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/files/6911790/Wireless.USB.Big.Sur.Adapter-V-4.zip)

**Reply by Tockman** [reply]
OK, thanks. I'll try it tomorrow.

**Reply by chris1111** [reply]
Uninstall the previous release  then try this one

**Reply by Tockman** [reply]
Sure. Never did it another way.

**Reply by chris1111** [reply]
send me the old package AWUS 1900 or the driver and utility using for catalina

**Reply by Tockman** [reply]
https://www.mediafire.com/file/mwqxvrfv3xii0by/ALFA_USB_WiFi_Driver_%2528Build_0919%2529.pkg/file

**Reply by chris1111** [reply]
We have also RtWlanDisk.kext on this program maybe thats the trick ?? I hope
[Wireless USB Big Sur Adapter.zip](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/files/6911811/Wireless.USB.Big.Sur.Adapter.zip)

**Reply by Tockman** [reply]
Maybe. Will give it a try. Thanks!

**Reply by chris1111** [reply]
> Maybe. Will give it a try. Thanks!

I think this is work for Alfa awus 1900

**Comment by chris1111** [comment]
HA ha  its work with a TL WN821N 
![Screen Shot 2021-07-31 at 5 19 24 PM](https://user-images.githubusercontent.com/6248794/127752675-9d46eff1-e769-4ba1-89a1-95951a4df319.png)

**Reply by Tockman** [reply]
And TL-WN821N is USB 2.0 802.11N adapter.

**Reply by chris1111** [reply]
> And TL-WN821N is USB 2.0 802.11N adapter.

Yes its old
see drivers install without issue working great
![Screen Shot 2021-07-31 at 5 29 59 PM](https://user-images.githubusercontent.com/6248794/127752852-a51bbcbc-b090-40ed-bfb9-357d77788428.png)

Probably working on other I make test I have 5 more adapter

**Comment by chris1111** [comment]
Test Working
TL WN722N V2
edimax ew 7722utn v2
TL-WN823N V3
Edimax Wi-Fi N150

**Comment by Tockman** [comment]
No, no, no :( Ain't no sunshine. Neither v4, nor special one with RtWlanDisk.kext - no work.
I feel the single solution left is to install last supported macOS (in my case 10.15.7) and original driver.
Thank you one more time.

**Reply by chris1111** [reply]
> No, no, no :( Ain't no sunshine. Neither v4, nor special one with RtWlanDisk.kext - no work.
> I feel the single solution left is to install last supported macOS (in my case 10.15.7) and original driver.
> Thank you one more time.

Do you use this
[Wireless USB Big Sur Adapter.zip]( https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/files/6911811/Wireless.USB.Big.Sur.Adapter.zip)
This is the program adapted with you package AWUS 1900

**Reply by Tockman** [reply]
Is it something different than you offered [here](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/discussions/130#discussioncomment-1095182)?

**Reply by chris1111** [reply]
> Is it something different than you offered [here](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/discussions/130#discussioncomment-1095182)?

No this is the same

**Reply by Tockman** [reply]
I refer to it as "special one with RtWlanDisk.kext".

**Reply by chris1111** [reply]
> I refer to it as "special one with RtWlanDisk.kext".

Yes

**Reply by Tockman** [reply]
https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/discussions/130#discussioncomment-1111662

**Comment by chris1111** [comment]
https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/discussions/130#discussioncomment-1095144

**Reply by Tockman** [reply]
I've tried everything you gave. Result is negative :(

**Comment by chris1111** [comment]
With OC ?

**Reply by Tockman** [reply]
But you told me OC's not my case?

**Comment by chris1111** [comment]
i never told you this, you told me posting in wrong thread and I sais special program exist also for OC

**Reply by Tockman** [reply]
https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/discussions/130#discussioncomment-1094223
I asked you: _"Just a dumb control question: if I'm on an unsupported Mac using Monterey with OCLP, should I use "OC" or "non-OC" version of utility? Is that a Hackintosh, technically?"_
Your answer was: _"for non-OC! you suppose to have this message..."_

**Comment by chris1111** [comment]
Thats nothing to do with  **(not my case?)**

**Reply by Tockman** [reply]
I really don't know what way your utility would interact with Open Core Legacy Patcher. So all that's left for me is to ask and believe ;)

**Comment by chris1111** [comment]
Open Core Legacy Patcher  Installe OC  in EFI partition this is correct

**Reply by Tockman** [reply]
Yes. So, once more, do I need an OC or non-OC version of your utility while I already have OC in my EFI?

**Reply by chris1111** [reply]
> Yes. So, once more, do I need an OC or non-OC version of your utility while I already have OC in my EFI?


If you use Opencore to boot your macOS you can use the OC version of this program but the non-OC version is supposed to work

**Reply by Tockman** [reply]
Well. I'll check this hypothesis next weekend. All I have up to this moment is non-OC version doesn't install driver (or doesn't let inject it) but it lets network utility load and run correctly.

**Reply by Tockman** [reply]
Utility = StatusBarApp, I mean.

**Reply by chris1111** [reply]
> Utility = StatusBarApp, I mean.


The network utility would not work at all if the driver was not loaded.

**Reply by Tockman** [reply]
So that's a mystery!

**Reply by Tockman** [reply]
You saw my screenshot. macOS sees my NIC as disconnected or unplugged. Usually this happens when driver didn't install correctly.

**Reply by Tockman** [reply]
Or maybe that's an USB failure, as I meant before. But since my USB ports works normally in virtual machine I can conclude it's a software problem, not hardware one.

**Comment by chris1111** [comment]
This is what happend when drivers not load
![USBWiFi_Unplug](https://user-images.githubusercontent.com/6248794/127770387-6d2353b8-c43a-49d5-ab7f-bbdd0fcb06bf.png)


what i suggest to you so as not to get confused is to delete all the programs you have because you seem to be using a bunch of stuff and it is confusing. So when you go to try a program you test it and if it doesn't work you uninstall it, reboot macOS and put it in the trash. Then you try another one and so on

**Reply by Tockman** [reply]
You're right, but... that's my working computer. And the adapter I bought was bought for working, also. So right now I can not afford to uninstall all the programs I have there. More simple is to make a partition for supported OS, install NIC and then, if successfully, migrate all my stuff there.

**Reply by chris1111** [reply]
> You're right, but... that's my working computer. And the adapter I bought was bought for working, also. So right now I can not afford to uninstall all the programs I have there. More simple is to make a partition for supported OS, install NIC and then, if successfully, migrate all my stuff there.


you misunderstood my answer, when I talk about uninstalling I'm talking about Wireless USB Adapetr and nothing else

**Reply by Tockman** [reply]
But I uninstalled it after every try! It's obvious and clear, I think.

**Reply by Tockman** [reply]
During last night and day I rebooted 28 times. It's more than a yearly norm ;)

**Reply by chris1111** [reply]
> During last night and day I rebooted 28 times. It's more than a yearly norm ;)


I tell you to uninstall it and also delete it so as not to get confused because if you have several it can lead to confusion

**Reply by Tockman** [reply]
And be sure, I made it too. Why keep the version that doesn't suit?

**Reply by chris1111** [reply]
> And be sure, I made it too. Why to keep the version that doesn't suit?

Ok its just to be sur because i have seen people mix in this way i just wanted to massage you to do it

**Reply by Tockman** [reply]
So will try OC version next time. Who knows, maybe it will work?

**Reply by chris1111** [reply]
> So will try OC version next time. Who knows, maybe it will work?

I really hope

**Reply by Tockman** [reply]
Just one question, Chris: won't EFI made by Wireless-USB-OC-Big-Sur-Adapter, overwrite existing EFI made by Open Core Legacy Patcher?

**Comment by chris1111** [comment]
No no it just enable the kext in config.plist then install kext in OC

**Comment by Tockman** [comment]
I've got two news, Chris.
The good one is OC version has installed fully and completely! Hallelujah...
And the bad one - my NIC connects to USB 2.0 only and, therefore, can not provide connection speed more than 2.0 standard presumes (~250 Mb/s). I know there's a special section in Linux driver to point device to connect only to 3.0 port. Is it possible to implement something like that?

And thank you, thank you, thank you again!

**Reply by Tockman** [reply]
And yes - I tried all 9 USB ports I have. All of them support USB 3.0 and all of them give 2.0 to my AWUS 1900.

**Comment by chris1111** [comment]
there is no source of the utility and the drivers to implement some thing like this, is suppos to be work USB 3 
if not its probably relate to the new macOS this drivers is made for 10.15 and for two years people enjoy new mac system with it

**Reply by Tockman** [reply]
https://github.com/aircrack-ng/rtl8814au
Here, at the bottom of the page.

**Comment by chris1111** [comment]
usb 3 is working for other device?

**Reply by Tockman** [reply]
Yes, all my external disks are connected by 3.0.

**Reply by Tockman** [reply]
![Снимок экрана 2021-08-07 в 13 50 45](https://user-images.githubusercontent.com/74548419/128597722-99d7883a-c7d4-446e-8bb5-0b81f1d6cea4.png)

**Reply by Tockman** [reply]
It's same USB3.0 Satechi hub. But discs are connected 3.0 and the NIC - as 2.0. The same thing happens when I plug the NIC directly at Mac's USB connector.

**Comment by chris1111** [comment]
Sadly you are not the first one report USB 3 not working on wifi adapter on new macOS

**Reply by Tockman** [reply]
I read it can be cured by USB mapping. But I am extremely incompetent in that.

**Comment by Tockman** [comment]
There's something very strange happens with USB 3.0 in Monterey. I've just tried another USB adapter - for this time it's ASUS AC-68 which is 100% supported. I plugged it in (driver was already installed), it connected, looked for the network and... then icon in menu bar disappeared. After 3-5 seconds it appeared again, adapter continued to search the network and failed again. And so on and on.

I understand what's happened. Just like its predecessor Alfa AWUS1900, ASUS was connected to my iMac as USB 2.0 device. But, unlike AWUS, ASUS is pure USB 3.0 and can not work with USB 2.0. And it just could not...

I tried different ports, and no luck. It looks like Apple consciously makes obstacles for external USB devices in Monterey because they can be used as a substitute on unsupported Macs.

**Comment by Tockman** [comment]
@chris1111 
After long weeks of self-torturing and changing different cables' and hubs' types and sorts I, at last, got snatched AWUS 1900's consent to connect in USB 3.0 mode. Yes, it's connected, right now. But is it interesting to know what connection speed does it show with USB 3.0? The same damned 480 Mb/s! Just like in 2.0 mode! And it far ain't that USB 3.0 speed solemnly promised by its manufacturer. I'd like to know your opinion on that: may it be driver-related problem?
![Снимок экрана 2021-12-05 в 02 46 07](https://user-images.githubusercontent.com/74548419/144728018-ab67c6d0-643e-4b7e-bd1e-f207e25d1508.png)
?

**Reply by chris1111** [reply]
My opinion is that I am not surprised 
![2018316149](https://user-images.githubusercontent.com/6248794/144729618-28ff108c-e339-4c5d-8361-45af5e13ae97.png)

I still doubt the USB 3 speed on this Wifi adapter, maybe in  the old macOS like Yosemite or earlier it works but on newer ones not sure of the performance 😄 Better to laugh about it than to cry

**Comment by Tockman** [comment]
@chris1111 
But, you know, ASUS AC-68 do really the same speed!!
![Снимок экрана 2021-12-06 в 00 48 40](https://user-images.githubusercontent.com/74548419/144765342-e72daf20-63fe-40c9-8636-767602bd6141.png)

**Reply by Tockman** [reply]
Moreover, ASUS AC-68 doesn't want to connect to 5 GHz network - 2,4 only :( AWUS1900, at least, connected to 5 instantly.

**Reply by Tockman** [reply]
And, by the way, my router registers AC-68 as 802.11n device, not .ac one.

**Comment by Tockman** [comment]
@chris1111 
Regarding USB speed problem: I contacted ALFA support, and they pointed to the driver as its root (let's excuse them for some mistakes):
"Our RD team use the same driver of the Github and show the 480Mbps that had same issue.
I think it is driver issue, when we use the Windows OS PC that did not had same issue".

**Comment by chris1111** [comment]
Take a look here Alfa AWUS036ACM Mediatek 🤣  Speed is good Up to 5G
My new Utility here ➤ [Notarized-RT2870USBWirelessDriver](https://github.com/chris1111/Notarized-RT2870USBWirelessDriver)
<img width="428" alt="Screen Shot 2022-01-09 at 7 38 49 PM" src="https://user-images.githubusercontent.com/6248794/148707855-f3594e55-d582-4f58-9e85-d4ab5111a49f.png">

**Reply by Tockman** [reply]
Thank you for the suggestion! It looks very promising. Single question: does it work with Mediatek chipset based adapters only? Because both of my USB adapters are Realtek.

**Comment by chris1111** [comment]
Realtek not work with Mediatek, my new Utility works SIP enable or Sip disable
To make work SIP disable your only need the secound part `Install.command` then plug USB adapter with a reboot.

I am  Buying and using [Alfa AWUS036ACM](https://www.amazon.ca/Alfa-AWUS036ACM-Long-Range-Dual-Band-Wireless/dp/B073X6RL9D)

**Reply by Tockman** [reply]
I see. To buy third adapter in my case looks a bit redundant so I'll give a try your new utility :)

**Reply by Tockman** [reply]
>  then plug USB adapter with a reboot.
You mean "with a reboot in recovery mode" or just reboot? ;)

**Reply by Tockman** [reply]
Now I have this:
![Снимок экрана 2022-01-11 в 19 36 52](https://user-images.githubusercontent.com/74548419/148983652-7f2025b5-8f06-4b81-801d-066ea08660da.png)
And I'm afraid if enabled SIP I'll get some problems with OCLP boot, because it required SIP disabled.

**Comment by chris1111** [comment]
if your setting is not good the script d'ont let you to continue

**Reply by Tockman** [reply]
Let's repeat it once more: you said that my wireless USB should work much better if SIP is enabled? And even throughput of my USB port will increase?

**Reply by chris1111** [reply]
> Let's repeat it once more: you said that my wireless USB should work much better if SIP is enabled? And even throughput of my USB port will increase?


I have noticed that on my Macbook Pro which is on Monterey having the SIP activated and its working verry well

**Reply by Tockman** [reply]
But my SIP _is_ enabled.
![Снимок экрана 2022-01-12 в 02 44 22](https://user-images.githubusercontent.com/74548419/149038793-8ecd57aa-80c5-473b-90e4-ba6c424ab7c1.png)

**Comment by chris1111** [comment]
SIP Enable

![Screen Shot 2022-01-11 at 6 50 43 PM](https://user-images.githubusercontent.com/6248794/149039450-6195d03b-3bfc-4356-8d32-d4813425dd24.png)

**Reply by Tockman** [reply]
**Ausdauersportler**: "_SIP is only partially disabled with OCLP when needing patches, but it will break booting a patched stream when enabling it. So leave it disabled!_"

**Comment by chris1111** [comment]
I know thats Why I am not using OCLP on my [MBP ](https://github.com/chris1111/MacBookPro-retina-late-2013)😄 Anyway 
To make work SIP disable your only need the secound part Install.command then plug USB adapter with a reboot.

**Reply by Tockman** [reply]
I understand but if I do it I just won't boot my OS! Because I'm on unsupported Mac (see above).

**Comment by Tockman** [comment]
Today after installing macOS 12.4 b4 your driver gave up to load - for the first time since install. By the way, does system extension by Marvell Semiconductor relate to some component of the driver? Because I got a message of its blocking.

**Comment by Tockman** [comment]
Reinstallation didn't help. It seems that OS just prevents Realtek kext from loading.
RtWlanU1827 and RtWlanU are still in Extensions folder but there's no more USB adapter among network devices.

**Comment by chris1111** [comment]
I was surprised by your comment so I rushed to test it
you are wrong my friend
Iam on  MacBookPro-retina-late-2013
With out OCLP, only SIP disable for test it

I will test SIP Enable next post

<img width="717" alt="Screen Shot 2022-05-03 at 8 38 33 PM" src="https://user-images.githubusercontent.com/6248794/166609478-f5157e3f-e4f4-4e41-88e2-7229eeecb711.png">
<img width="406" alt="Screen Shot 2022-05-03 at 8 43 47 PM" src="https://user-images.githubusercontent.com/6248794/166609210-ca1f1100-208b-46da-9b6c-5eca2b140f61.png">
<img width="780" alt="Screen Shot 2022-05-03 at 8 44 14 PM" src="https://user-images.githubusercontent.com/6248794/166609212-3c671771-9125-4047-8513-8ee55963fea4.png">

**Comment by chris1111** [comment]
[SIP Enable also works](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/discussions/159)

<img width="817" alt="Screen Shot 2022-05-03 at 8 55 46 PM" src="https://user-images.githubusercontent.com/6248794/166610027-cd163b4f-dcf4-4476-91a6-f8248ebb5bd1.png">

**Comment by Tockman** [comment]
Thank you for your fast reaction!
So may be that's a kind of interference with OCLP (0.4.4, in my case)?

**Comment by Tockman** [comment]
Tried to reinstall once more - no luck. After installation there's no demand to confirm permission for Realtek drivers. I think, that's because they are already in a trusted list. Drivers are in Extensions folder. Icon is here in dock but red-crossed. May be, the reason is installer sees SIP as disabled, but in fact it's enabled, with just SSV turned-out?
![Снимок экрана 2022-05-04 в 10 43 38](https://user-images.githubusercontent.com/74548419/166640708-82e92650-bc54-4baa-9398-ac8a43839fca.png)

**Comment by Tockman** [comment]
And what if I copy/replace RtWlanU1827 and RtWlanU to /EFI/OC/kexts folder? Will they load at startup then?

**Comment by chris1111** [comment]
You here [Wireless USB Big Sur Adapter](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter) there is no OC or Clover here

[SIP Enable also works](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/discussions/159)

`Note: You can Enable and Disable the SIP from OpenCore, It works the same way.`

**Reply by Tockman** [reply]
> You here [Wireless USB Big Sur Adapter](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter) there is no OC or Clover here

Yes, I know. But this very version worked on my machine in the run of whole year. And now it won't load, and I ask you what I'd do.
Also I was strongly advised not to enable SIP permanently (https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/discussions/130#discussioncomment-1951438). My macOS runs with OpenCore Legacy Patcher, so I asked if there is some reason to place kext files into EFI.

**Comment by chris1111** [comment]
kext file in EFI  for OpenCore works wit this but SIP disable
But your not need to place any kexts, the program do it for you
[Wireless-USB-OC-Big-Sur-Adapter](https://github.com/chris1111/Wireless-USB-OC-Big-Sur-Adapter)

**Comment by Tockman** [comment]
Look what I found! System Report > Extensions: Loading not supported for both kexts.
![Снимок экрана 2022-05-04 в 12 44 23](https://user-images.githubusercontent.com/74548419/166659074-64958569-27f5-4395-be50-8f2b1fa07f9b.png)
![Снимок экрана 2022-05-04 в 12 44 38](https://user-images.githubusercontent.com/74548419/166659084-5f979dca-b118-4671-a93b-217ac0374723.png)
Third line from bottom "Возможна загрузка: Не поддерживается" means "Possibility of load: Not supported".

**Comment by chris1111** [comment]
I am not use OCLP you have to use the program [Wireless-USB-OC-Big-Sur-Adapter](https://github.com/chris1111/Wireless-USB-OC-Big-Sur-Adapter) from the readme SIP must be disable

**Reply by Tockman** [reply]
But it is disabled! And I used to have Wireless-USB-Big-Sur-Adapter (non-OC) before yesterday, and it worked normally. Now it runs through installation, but kexts won't load. OK, if you suggest that now I have to try OC version, I'll give it a try.

**Comment by chris1111** [comment]
both way works from all tree program 
Enable/Disable
follow the guide corectly

**Comment by chris1111** [comment]
You seem to mix the programs you want to use the kexts injections Opencore so dont use the one which is designed for the Library/Extensions.

**Reply by Tockman** [reply]
Yes, you were right - OC version installed kexts and got working. Thank you for your time!

**Comment by Tockman** [comment]
@chris1111, I've just realized that, due to some changes made to OCLP architecture (in particular, offline install mode only available), during OCLP upgrade (from 0.4.4 to 0.4.5) all previous installed kexts have been wiped out from EFI/OC/Kext, and I had to reinstall Wireless USB OC Big Sur to get my adapter working. I'm seeing it for the second time now. So, is reinstall the single possibility, or may be there's any solution to keep kexts where they are? Thank you.

**Comment by chris1111** [comment]
Yes you can try to make a copy of the both kexts then rename it `RtWlanU`
`RtWlanU1827` it will apear at folder, after OCLP update rename it at RtWlanU.kext
RtWlanU1827.kext
I dont think OCLP delete the two folder 😁 But chek your config.plist if the kext is enable, if not enable the both kexts

**Reply by Tockman** [reply]
Hi chris,

since there's no dedicated discussion thread for D-LinkUtility-Package I allow myself to write it there. At last I've bought AWUS036ACM (based on Mediatek chip) and installed your driver for it. Good news is that adapter is really displayed as USB3 device (up to 5 Gb/sec). 
![Снимок экрана 2022-07-22 в 21 19 05](https://user-images.githubusercontent.com/74548419/180501044-c7cafefb-45f4-4b18-8ca9-402783b464f1.png)
And the bad one - it's seen by Monterey as 802.11N device, not 802.11AC one. Accordingly, the connection speed is limited to 300 Mb/sec, and that's very poor for Gigabyte network. 
![Снимок экрана 2022-07-22 в 21 24 07](https://user-images.githubusercontent.com/74548419/180501091-5fc2f2d1-870d-4329-8978-d158840638a5.png)
Are there some thoughts on how it might be fixed? Thank you!

**Reply by chris1111** [reply]
Yea I told you about the Mediatek chip its support USB 3 but this Drivers is not really full stable, some time it can disconected then no solution exist for anny issue about this chip.
